Added basic bplist17 support (Tested on iOS 14.6)

weizenberg opened the pull request:
I added bplist17 (and 16 as well) parsing support:

I used the XPCDecoder class to load the xpc message and decoding it, so it should support all the formats (not tested yet).
Currently, I'm just printing the debugDescription which giving a nice picture of what going on, but we can also run over the loaded classes and try to print their description using toString method. 

I tested this on iOS 14.6, you're welcome to test on other iOS versions.

weizenberg committed:
Added basic bplist17 support (Tested on iOS 14.6)

hot3eed commented:
Awesome, thanks! Will review this weekend.

hot3eed commented:
Hey, sorry for the delay on this. Do you happen to know daemons that use bplist17 to test with?

weizenberg commented:
Hey, sure no problem!

Almost any system daemon from 14 and up use bplist17, because it all sharing the same CF framework. You can check this with accountsd daemon.

Commit:
bplist: Fallback to NSXPCDecoder API (+ refactors)

Use the NSXPCDecoder API as a generic bplist parser for formats for
which we don't have specific parsers. I.e. bplist 16 and bplist17.

hot3eed commented:
Apologies again for the delay on this. Just go this PR landed in [v0.8.1](https://github.com/hot3eed/xpcspy/releases/tag/v0.8.1). Thanks!
